File: src/main/java/ru/dushkinmir/absolutelyRandom/core/CommandsManager.java
package ru.dushkinmir.absolutelyRandom.core;

import dev.jorel.commandapi.CommandAPI;
import org.bukkit.plugin.Plugin;
import ru.dushkinmir.absolutelyRandom.features.sex.SexCommands;
import ru.dushkinmir.absolutelyRandom.features.warp.WarpCommands;

public class CommandsManager {
    private final Plugin plugin;
    private final ExtensionManager extensionManager;

    public CommandsManager(Plugin plugin, ExtensionManager extensionManager) {
        this.plugin = plugin;
        this.extensionManager = extensionManager;
    }

    public void registerCommands() {
        // Initialize and register WarpCommandManager
        WarpCommands wcm = new WarpCommands(extensionManager.getWarpManager(), plugin);
        // Initialize and register SexCommandManager
        SexCommands scm = new SexCommands(extensionManager.getFissureHandler(), plugin);
        wcm.registerWarpCommands(); // Register warp commands
        scm.registerSexCommand(); // Register sex commands
        CommandAPI.onEnable(); // Enable CommandAPI
    }

    public void unregisterCommands() {
        // Unregister commands
        CommandAPI.unregister("debugrandom");
        CommandAPI.unregister("warp");
        CommandAPI.unregister("sex");
        plugin.getLogger().info("Команды CommandAPI отменены.");
        // Disable CommandAPI
        CommandAPI.onDisable();
    }
}

File: src/main/java/ru/dushkinmir/absolutelyRandom/core/CommandsManager.kt
package ru.dushkinmir.absolutelyRandom.core

import dev.jorel.commandapi.CommandAPI
import org.bukkit.plugin.Plugin
import ru.dushkinmir.absolutelyRandom.features.sex.SexCommands
import ru.dushkinmir.absolutelyRandom.features.warp.WarpCommands

class CommandsManager(private val plugin: Plugin, private val extensionManager: ExtensionManager) {

    fun onEnable() {
        // Initialize and register WarpCommandManager
        val wcm = WarpCommands(extensionManager.getWarpManager(), plugin)
        // Initialize and register SexCommandManager
        val scm = SexCommands(extensionManager.getFissureHandler(), plugin)
        wcm.registerWarpCommands() // Register warp commands
        scm.registerSexCommand() // Register sex commands
        CommandAPI.onEnable() // Enable CommandAPI
    }

    fun onDisable() {
        // Unregister commands
        CommandAPI.getRegisteredCommands().forEach { CommandAPI.unregister(it.commandName) }
        plugin.logger.info("All commands are deregistered.")
        // Disable CommandAPI
        CommandAPI.onDisable()
    }
}